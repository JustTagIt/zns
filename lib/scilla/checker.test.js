const fs = require('fs')
const path = require('path')

const checker = require('./checker')
const paths = require('../paths')

it('should pass a valid scilla file', () => {
  expect(
    checker({
      input: path.join(__dirname, '../example_contracts/hello.scilla'),

      cashflow: true,
    }),
  ).toMatchSnapshot()
})

it('should throw on error scilla file', () => {
  expect(() => {
    checker({
      input: path.join(__dirname, '../example_contracts/invalid.scilla'),
    })
  }).toThrowErrorMatchingSnapshot()
})

it("should throw on warning if in 'strict mode' scilla file", () => {
  expect(() => {
    checker({
      input: path.join(__dirname, '../example_contracts/hello.scilla'),

      strict: true,
    })
  }).toThrowErrorMatchingSnapshot()
})
