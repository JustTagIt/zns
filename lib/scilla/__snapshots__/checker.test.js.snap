// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should pass a valid scilla file 1`] = `
Object {
  "cashflow_tags": Object {
    "ADT constructors": Array [],
    "State variables": Array [
      Object {
        "field": "owner",
        "tag": "NotMoney",
      },
      Object {
        "field": "welcome_msg",
        "tag": "NoInfo",
      },
    ],
  },
  "warnings": Array [
    Object {
      "end_location": Object {
        "column": 0,
        "file": "",
        "line": 0,
      },
      "start_location": Object {
        "column": 3,
        "file": "/Users/laptop3/ZilliqaNameService/contracts/lib/example_contracts/hello.scilla",
        "line": 54,
      },
      "warning_id": 1,
      "warning_message": "Unable to validate that send arguments is null or single message",
    },
    Object {
      "end_location": Object {
        "column": 0,
        "file": "",
        "line": 0,
      },
      "start_location": Object {
        "column": 0,
        "file": "",
        "line": 0,
      },
      "warning_id": 1,
      "warning_message": "No transition in contract HelloWorld contains an accept statement
",
    },
  ],
}
`;

exports[`should throw on error scilla file 1`] = `
"{
  \\"errors\\": [
    {
      \\"error_message\\": \\"Syntax error.\\",
      \\"start_location\\": {
        \\"file\\": \\"/Users/laptop3/ZilliqaNameService/contracts/lib/example_contracts/invalid.scilla\\",
        \\"line\\": 1,
        \\"column\\": 8
      },
      \\"end_location\\": {
        \\"file\\": \\"\\",
        \\"line\\": 0,
        \\"column\\": 0
      }
    }
  ],
  \\"warnings\\": []
}"
`;

exports[`should throw on warning if in 'strict mode' scilla file 1`] = `
"{
  \\"warnings\\": [
    {
      \\"warning_message\\": \\"Unable to validate that send arguments is null or single message\\",
      \\"start_location\\": {
        \\"file\\": \\"/Users/laptop3/ZilliqaNameService/contracts/lib/example_contracts/hello.scilla\\",
        \\"line\\": 54,
        \\"column\\": 3
      },
      \\"end_location\\": {
        \\"file\\": \\"\\",
        \\"line\\": 0,
        \\"column\\": 0
      },
      \\"warning_id\\": 1
    }
  ]
}"
`;
